src/decays/flexibledecay_settings.cpp

```diff
@@ -120,7 +120,7 @@ void FlexibleDecay_settings::set(Settings o, double value)
    case include_higher_order_corrections: // 2 [int >= 0 and <= 4]
       assert_integer(value, descriptions.at(o).c_str());
       assert_ge(value, 0, descriptions.at(o).c_str());
-      assert_le(value, 4, descriptions.at(o).c_str());
+      assert_le(value, 41, descriptions.at(o).c_str());
```

Comment:
I think this assertion would not fire if `value == 39`, which would not be a valid value. I think it would be better to check that the 0th digit is within [0,1] and the 1st digit is within [0,4] (intervals of integers). Maybe write something along the lines
~~~.cpp
assert_le(value % 10, 1, descriptions.at(o).c_str());
assert_le((value/10) % 10, 4, descriptions.at(o).c_str());
~~~

Reply:
Maybe something like this?
```cpp
assert_ge(value, 0, descriptions.at(o).c_str());    
assert_le(static_cast<int>(value) % 10, 1, descriptions.at(o).c_str());    
assert_le(static_cast<int>(value/10) % 10, 4, descriptions.at(o).c_str());    
assert_le(static_cast<int>(value) % 10, static_cast<int>(value/10) % 10, descriptions.at(o).c_str());
```
It ensures that SM is limited to N^4LO, BSM to NLO and that BSM order is always lower or equal to SM so that we don't compute BSM corrections without the SM ones.